design/kiota-e2e.md

```diff
@@ -23,8 +23,155 @@ Before we jump into the end-to-end walk-through, it's important to set some cons
 
 ## Table of Contents
 
--   [NodeJS e2e using the Service library](#node-service-library)
--   [NodeJS e2e using the Core library](#node-core-library)
+- [Foudation](#foundation)
+- [NodeJS e2e using the Service library](#node-service-library)
+- [NodeJS e2e using the Core library](#node-core-library)
+
+## <a id="foundation"></a> Foundation
+
+This section lists all the foundation elements of the Graph JS SDK and how it affects the developer's experience.
+
+### Models
+
+The models should be easily consumable by developers in a natural format. For JavaScript and TypeScript developers, interfaces are the most common way to understand, represent and send data to endpoints.
+
+Thanks to the TypeScript nature of allowing merging of Classes and Interfaces with the same name, models are represented in a way that allows both rich capabilities of the classes (backing stores, dirty tracking, etc.) and simplicity of using interfaces. 
+
+```typescript
+export class SendMailRequestBody implements Parsable, SendMailRequestBody {
+	/** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.  */
+    private _additionalData: Map<string, unknown>;
+    private _message?: Message | undefined;
+    private _saveToSentItems?: boolean | undefined;
+    /**
+     * Instantiates a new sendMailRequestBody and sets the default values.
+     */
+    public constructor(value?: Partial<SendMailRequestBody>) {
+        if(value) {
+            Object.assign(this, value);
```

Comment:
this section might be a bit more complex than that, we need to account for properties that are complex types (message body) or collections of complex types (message to recipients) as well.

Reply:
Not the goal. I Think I'll simply remove the code. This is not meant for the internal implementation, but the way developers use the SDK.

Reply:
yep, replace by a comment saying the actual implementation object gets instantiated from the partial values